Sort switch fzf by recency

ChrisPenner opened the pull request:
* [x] Merge #5553  first

## Overview

With  the addition of #5553  it's now possible to list projects and branches by which were most recently visited.

This change lists project branches putting most recently accessed first.

## Implementation notes

* Sort fzf completion list so it goes: 
  * project branches in order of most recently accessed,
  * then the  list of all projects

## Interesting/controversial decisions

I  removed the `/` prefixing thing since it didn't really make sense to include names twice like I was when sorting in this new way, you'd always see the shortcut name next to the full name, and I think it's important to keep the full name in  case the user types the project name and expects to find it that way.

## Test coverage

See fzf  transcript

## Loose ends

Maybe we'll want to tweak it after trying it out a bit.

ChrisPenner committed:
Sort project branches by recency in switch fzf

aryairani commented:
The windows transcripts complained about a branch order mismatch; and it looks like it was [already merged](https://github.com/unisonweb/unison/actions/runs/12934962331/job/36077962939?pr=5555#step:2:81) with https://github.com/unisonweb/unison/commit/5d2aa196966c73b0b60ef038120d3a81f576a1ef when it failed, so I'm kind of confused about why it failed. But it passed on a re-run, which is also "whatever".

I can go ahead and merge it, since our policy has been to merge things that sometimes pass, but do you want to note anything about the failure vs the determinism fix?

ChrisPenner committed:
Fix flakiness in output branch order

the branches must come alphabetically BEFORE main since they'll be more recently accessed.

ChrisPenner commented:
@aryairani , Hrmm, I think I found what's happening here; determinacy is hard 😓 

I think I actually need the branch names to come alphabetically _before_ main to get determinacy, since all the other branches will be created _after_ 'main' is; 
Looking at the transcript runner it doesn't call `switch` unless the next stanza is in a _different_ project-branch, so I believe switching to assuming 'main' will have the _oldest_ last-accessed time should be valid.

ChrisPenner commented:
@aryairani seems better? At least on the last run :P

aryairani commented:
timestamp field has a 1-second granularity currently, so we have to use this One Weird Trick to produce the same result regardless of whether each step completes in under a second (fallback to name) or over a second (fallback to timestamps)

idea: use a more granular timestamp in the future
question: if we try to use a more granular timestamp, can sqlite even produce distinct timestamps within a single transaction?
